File: c++/tools/fib.cpp
#include <iostream>
#include <string>

int fib(int count) {
	int prev = 0;
	int curr = 1;

	for(int i = 0; i < count; i++) {
		int tmp = curr;
		curr += prev;
		prev = tmp;
	}

	return prev;
}

int main(int argc, char* argv[]) {
	if (argc < 1) {
		std::cout << "You need to supply an argument\n";
		std::exit(1);
	}

	int count = atoi(argv[1]);

	if (count < 0) {
		std::cout << "You need to use a positive number\n";
		std::exit(1);
	}

	int result = fib(count);

	std::cout << result << std::endl;

	return 0;
}

File: go/tools/fib.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func fib(count int) int {
	prev := 0
	curr := 1

	for i := 0; i < count; i++ {
		tmp := curr
		curr += prev
		prev = tmp
	}

	return prev
}

func main() {
	if len(os.Args) < 1 {
		fmt.Println("You need to supply an argument")
		os.Exit(1)
	}

	countStr := os.Args[1]

	count, err := strconv.Atoi(countStr)
	if err != nil {
		fmt.Printf("Please enter a number, you entered %s\n", countStr)
		os.Exit(1)
	}

	if count < 0 {
		fmt.Println("You need to use a postive number")
		os.Exit(1)
	}

	result := fib(count)

	fmt.Println(result)
}
